src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,171 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.HashMap;
+
+import command.AddTaskCommand;
+import command.ClearCommand;
+import command.Commandable;
+import command.DeleteCommand;
+import command.FindCommand;
+import command.HelpCommand;
+import command.ListCommand;
+import command.MarkCommand;
+import command.ShutdownCommand;
+import dukeexception.InvalidCommandException;
+import dukeexception.InvalidVarException;
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.ToDo;
+
+
+/**
+ * Parses inputs passed via the user interface into commands if possible.
+ */
+
+public class Parser {
+    private HashMap<String, Commandable> stringToCommand;
+
+    /**
+     * Initializes the parser.
+     */
+    public Parser() {
+        stringToCommand = new HashMap<>();
+        init();
+    }
+    private void init() {
+        stringToCommand.put("bye", new ShutdownCommand());
+        stringToCommand.put("help", new HelpCommand());
+        stringToCommand.put("list", new ListCommand());
+        stringToCommand.put("todo", new AddTaskCommand());
+        stringToCommand.put("deadline", new AddTaskCommand());
+        stringToCommand.put("event", new AddTaskCommand());
+        stringToCommand.put("delete", new DeleteCommand());
+        stringToCommand.put("mark", new MarkCommand(true));
+        stringToCommand.put("unmark", new MarkCommand(false));
+        stringToCommand.put("find", new FindCommand());
+        stringToCommand.put("clear", new ClearCommand());
+    }
+
+    /**
+     * Parses the string passed to it and produces the corresponding command.
+     * @param input the string passed to the parser that is meant to be interpreted.
+     * @return the command that the string represents.
+     * @throws InvalidCommandException if the command cannot be identified.
+     * @throws InvalidVarException if the command is identifiable but the parameters are incorrect.
+     */
+    public Commandable parse(String input) throws InvalidCommandException, InvalidVarException {
```

Comment:
Generally, it seems that it is not recommended for methods to be too long (more than 30 lines). Perhaps some of the code could be abstracted out!